src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -26,41 +27,47 @@ public override object ConvertTo(ITypeDescriptorContext ctx, CultureInfo ci, obj
 
         public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, object data)
         {
-            object result;
+            object result = null;
+            bool valueIsInvalid = true;
 
             // For any error, throw the ArgumentException with SR.Invalid_enum_value
             try
             {
-                string value = (string)data;
-                if (string.IsNullOrEmpty(value)) throw new Exception();
-
-                // Disallow numeric values for enums.
-                if (!string.IsNullOrEmpty(value) &&
-                    (char.IsDigit(value[0]) ||
-                    (value[0] == '-') ||
-                    (value[0] == '+')))
-                    throw new Exception();
-
-                if (value != value.Trim())
+                if ((data is string value) && (value.Length > 0))
                 {
-                    // throw if the value has whitespace 
-                    throw new Exception();
+                    // Disallow numeric values and whitespace at start and end.
+                    if ((!char.IsDigit(value[0])) && (value[0] != '-') && (value[0] != '+') &&
+                        (!char.IsWhiteSpace(value[0])) && (!char.IsWhiteSpace(value[value.Length - 1])))
+                    {
+                        result = Enum.Parse(_enumType, value);
```

Comment:
Since this is the only place that could still throw an exception, you could get rid of the outer `try` and make this:

```C#
try
{
    return Enum.Parse(_enumType, value);
}
catch
{
    // Exception from parse. Will throw more appropriate exception below.
}
```
Then there's no need for `valueIsInvalid`. Anything that gets to that point was some sort of error, and the `throw CreateExceptionForInvalidValue();` is the thing to do.